table-api/src/main/java/io/deephaven/api/agg/AggregationOptimizer.java

```diff
@@ -111,4 +108,10 @@ public void visit(Partition partition) {
         visitOrder.computeIfAbsent(partition.includeGroupByColumns() ? PARTITION_KEEPING_OBJ : PARTITION_DROPPING_OBJ,
                 k -> new ArrayList<>()).add(partition.column());
     }
+
+    @Override
+    public void visit(Formula formula) {
+        // Supplying a `null` entry value indicates that the key is already an aggregation.
+        visitOrder.putIfAbsent(formula, null);
+    }
```

Comment:
This implementation does not seem like it matches up with the rest; I see `io.deephaven.api.agg.AggregationOptimizer#build` shows _some_ kind of support for null value, but I don't know how that would have even been possible before...

Reply:
I can't recall *why* I added that case before, but as a way to handle "do not further optimize this aggregation" it seems perfectly reasonable. I think Larry should continue down this path as-is, unless you have a counterproposal.